### DISCUSSION TITLE
is there a similar python-socketio call method functionality in Flask-socketio?

### DISCUSSION THREAD

**Discussion by alurmtr** [miguelgrinberg/Flask-SocketIO#1828] [Q&A]
I am using Flask-socketio and sending an emit to the client, and would like to wait for response before continuing in the code flow.

The case scenario is a background python code that needs feedback from user whether code should continue execution. An emit message is sent to the client, which in turns will send a js confirm pop-up window to the front end user. Based on the user selection, an acknowledgement is sent back to the python background process. 

The callback option in the flask-socketio emit wouldn't work well. Something like the python-socketio call method would be great to just wait for the response. just wondering if there is such an option in flask-socketio.

Thanks
Al

**Comment by miguelgrinberg** [comment]
I never got around to add `call()`, but I think there shouldn't be a problem exposing it here. There's always a little bit of extra complication in this extension due to the app/request contexts, but I don't think this should be any more complex than it already is for `emit()` and `send()`. I'll have a look within the next few days and see if I can add it.

**Reply by alurmtr** [reply]
Thanks for looking into it!. 

Al.
